providers/azure.go

```diff
@@ -120,70 +119,67 @@ func (p *AzureProvider) Redeem(redirectURL, code string) (s *sessions.SessionSta
 	}
 	err = json.Unmarshal(body, &jsonResponse)
 	if err != nil {
-		return
+		return nil, err
 	}
 
-	s = &sessions.SessionState{
+	return &sessions.SessionState{
 		AccessToken:  jsonResponse.AccessToken,
 		IDToken:      jsonResponse.IDToken,
 		CreatedAt:    time.Now(),
 		ExpiresOn:    time.Unix(jsonResponse.ExpiresOn, 0),
 		RefreshToken: jsonResponse.RefreshToken,
-	}
-	return
+	}, nil
 }
 
 func getAzureHeader(accessToken string) http.Header {
 	header := make(http.Header)
 	header.Set("Authorization", fmt.Sprintf("Bearer %s", accessToken))
+	header.Set("Accept", "application/json")
 	return header
 }
 
 func getEmailFromJSON(json *simplejson.Json) (string, error) {
-	var email string
-	var err error
-
-	email, err = json.Get("mail").String()
-
+	email, err := json.Get("mail").String()
 	if err != nil || email == "" {
```

Comment:
If you were to reverse this, you could return early here and then do the rest of the method under lesser indent
```
	if err == nil && email != "" {
            return email, nil
        }
```